src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunILLink.cs

```diff
@@ -454,6 +520,48 @@ private void EnableNonFrameworkTrimming(XDocument project)
                                                  new XElement("action"))));
         }
 
+        static readonly string substitutionsFilename = "ILLink.Substitutions.xml";
+
+        private void EmbedSubstitutions(XDocument project)
+        {
+            var ns = project.Root.Name.Namespace;
+
+            project.Root.Add (new XElement(ns + "ItemGroup",
+                                new XElement("EmbeddedResource",
+                                    new XAttribute("Include", substitutionsFilename),
+                                    new XElement("LogicalName", substitutionsFilename))));
+        }
+
+        private void AddFeatureSetting(TestProject testProject, string referenceAssemblyName)
+        {
+            // Add a feature definition that replaces Main. We use the existing InvariantGlobalization property
```

Comment:
> We use the existing InvariantGlobalization property

I'm not crazy about this .... it feels like we should make a new `@(RuntimeHostConfigurationOption)` item for this feature. And then in the `ILLink_ignores_host_config_settings_with_link_false` test case we can just add `Link=false` metadata to that item. That way we aren't hacking in the `DisableLinkingForHostConfig` method and hooking to a private target `_SetILLinkDefaults`. We would be testing like an actual user would be.

Reply:
The reason I did this is that I want the test to fail if the built-in `RuntimeHostConfigurationOption` definition ever moves to a target that doesn't run before the linker. Do you think that is worth testing for?

Reply:
I'm not sure I follow. All the built-in `@(RuntimeHostConfigurationOption)` definitions are in a global `ItemGroup`. Global ItemGroups are evaluated before Targets are executed.

If we ever moved the `@(RuntimeHostConfigurationOption)` definitions into a Target, that would be a breaking change.

Reply:
I see - I wasn't sure that that would be a breaking change. I will take your suggestion.